src/ontology/mondo-ingest.Makefile

```diff
@@ -3,7 +3,7 @@
 ## If you need to customize your Makefile, make
 ## changes here rather than in the main Makefile
 .PHONY: deploy-mondo-ingest build-mondo-ingest documentation mappings update-jinja-sparql-queries \
-report-mapping-annotations python-install-dependencies all-exclusions
+report-mapping-annotations python-install-dependencies slurp-all slurp-%
```

Comment:
ahy does this remove `all-exclusions`?

Reply:
I just rebased this `slurpdraft` branch from `main` after merging the "all/merged exclusions" PR.

The new command for this is no longer the `.PHONY` target `all-exclusions`. The target is now called `config/term_exclusions.txt config/exclusion_reasons.robot.template.tsv:`.

Reply:
Note to self: This sub-discussion is from 17 days ago. I can't remember if we agreed on these `.PHONY` targets before or after that, but either way, I remember that the last time we spoke about this, we agreed to have these phonies; so I need to add this back.

@matentzn I know this is not related to slurp, but I'll make this change either in this PR or directly to `main`. My question to you is: How many `.PHONY` targets do you want for the exclusion tables?

We have 3 distinct sets of artefacts that we're generating w/ the exclusion tables:
1. `$(REPORTDIR)/term_exclusions.txt $(REPORTDIR)/exclusion_reasons.robot.template.tsv`
2. `$(REPORTDIR)/term_exclusions.ttl`
3. `$(REPORTDIR)/excluded_terms_in_mondo_xrefs.tsv`

Which of these 3 do you want PHONY targets for? 1 for each? Or should I make one PHONY for (1-2) above combined, and a second one for (3)? And then perhaps an additional third target that runs (1-3)?

Reply:
i dont care so much about phony targets. A phony target is like a workflow. We only need it if we expect to run the task repeatedly. PHNOY goals should never be dependencies of other goals.

TLDR: as few phnonies as possible. But at least one for the whole workflow.

Reply:
I understand this is a bit of a sidetrack, but I'd still like to know some things.

> "at least one for the whole workflow"
Does "the whole workflow" mean all of `term_exclusions.txt`? And then don't worry about the `.ttl`? Or does it include the `.ttl`? Does it include doing the whole problematic xrefs analysis? Or are those separate workflows? If by "whole workflow" you mean (1-3) above, that's easy enough!

> PHONY goals should never be dependencies of other goals.
Not necessarily. I know of one exception. My Python scripts need the `pip` requirements to be installed. So I have made my phony `python-install-dependencies` a dependency of all of the make goals that use Python.

Reply:
> Not necessarily. I know of one exception. My Python scripts need the pip requirements to be installed. So I have made my phony python-install-dependencies a dependency of all of the make goals that use Python.

It is absolutely critical that your dependencies are at all times added to ODK the same moment they are added here (make PRs there). The `python-install-dependencies` should only be used at development time - after that, the ODK should take over..

> Does "the whole workflow" mean...

I could tell you my opinion for this specific case (one workflow to run everything (1-3 for all ontologies with one command), one workflow per-ontology), but the point is that there is no one correct answer to this question. Anything that can be usefully defined by a single command can be considered a workflow, and you will have to apply some judgement.

Reply:
#### depending on phony goals
I saw your comment in the commit message:

> 1. Never depend on a PHONY goal, because they force the whole pipeline to be rebuild - just add them to each goal, installing a few python dependencies that are already installed is fast enough.

This is very good for me to know, thanks. It makes perfect sense; should've thought of this. I'll put the running of PHONY goals within the block of execution for the target in the future; not as a prerequisite.

---

@matentzn Just so you know though, I think it is wrong that this forces the _whole pipeline_ to rebuild. What seems to happen is that it just forces the target itself to be rebuilt. I did a test:

```
.PHONY: xxx
xxx:
	@echo hi

tempy.txt: mirror/omim.owl xxx
	touch tempy.txt
```

The results of this: When I ran `make tempy.txt`, it did indeed force a re-run of the target. It did `touch tempy.txt`. However, it did not re-run `mirror/omim.owl`. It considered that up-to-date.

Reply:
You need to use `obolibrary/odkfull:dev`, I will have a new version up tomorrow (but the one up is fairly up to date).

Reply:
### Phony targets for workflows
Got it, so I can use my discretion here. Just a warning about giving me the liberty to create PHONY targets as workflow aliases. I have been known to be quite liberal in creating them for my own convenience. I can try to use my judgement to only write ones if I think they'll be useful for others, but I might be a bit biased to my preferences.